pkg/manager/nats_test.go

```diff
@@ -0,0 +1,324 @@
+package manager
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/kyma-project/nats-manager/pkg/k8s/chart"
+	chartmocks "github.com/kyma-project/nats-manager/pkg/k8s/chart/mocks"
+	k8smocks "github.com/kyma-project/nats-manager/pkg/k8s/mocks"
+	"github.com/kyma-project/nats-manager/testutils"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func Test_GenerateNATSResources(t *testing.T) {
+	t.Parallel()
+
+	// define test cases
+	testCases := []struct {
+		name         string
+		wantOwnerRef bool
+	}{
+		{
+			name:         "should work with empty options",
+			wantOwnerRef: false,
+		},
+		{
+			name:         "should apply the provided options",
+			wantOwnerRef: true,
+		},
```

Comment:
This is a bit confusing, I think it is better to have the givenOptions and expectedOptions test-case inputs.